scipio/src/parking.rs

```diff
@@ -93,15 +95,23 @@ impl fmt::Debug for Parker {
     }
 }
 
-struct Inner {}
+struct Inner {
+    wakers: RefCell<Vec<Waker>>,
```

Comment:
@glommer you want to use SmallVec instead of Vec? what capacity do you expect it to have on stack then?

Reply:
There are two options in my view.

The beautiful thing to keep in mind is this:

Note that we don't *have* to fetch all events from the ring buffer. Let's assume that reacting to each event takes 10us to process. Not the time to fetch the event, the time the user spends reacting to the event. For instance if you read something from the network, you will do something with it and send a response back, imagine that this think time takes on average 10us. 

That means that if you fetch 10,000 events, that is 100ms which is the maximum preempt timer setting.
What that means is that there is no situation on earth where it is ever worth it to read more than 10k events.

Events that are not consume can sit in the io_uring and we'll consume them next time.

seastar btw does something similar: Except for Seastar I used 128 as a hard limit because the preempt timer was less than 1ms.

Even for us, though, 10k is absurdly high: the io_uring itself only has a limited amount of buffer space. It will never ever generate 10k events in one row. We keep timers separately so it needs a bit more extra space, but still... even a 1k maximum would be enough.

Assuming every waker is 8 bytes ( a function pointer), that means 8kB stack space which is absolutely reasonable for the Reactor.

What to do with this info, though? There are. two choices:
* SmallVec with room for 1k or even 512 entries and if we ever have to read more, it will go on heap
* Our own data structure that has a maximum capacity for 1k or even 512 entries (backed by a static array) and has a `is_full` method. The io_uring implementation is changed such that before consuming a cqe we test for `is_full`: if we can't add more wakers that's fine, we just stop processing.

The second is a lot more work, I think, because the changes run deeper. But it would never trigger an allocation so it is better technically.

The first is simpler and to be honest, in practice I very very much doubt we'll ever have more than 1k wakers returning in one batch. We can probably just use SmallVec, which simplifies coding, and add a stat to the `ExecutorStats`, counting the number of times we went over the limit. Why complicate something before it is an issue?

I'll leave it up to you. To be honest, using a standard Vec works as well, so long as you reserve enough space for 1k entries in the beginning with `with_capacity`.